classes/Hook.php

```diff
@@ -885,7 +920,48 @@ public static function exec(
             }
         }
 
-        return $output;
+        if ($array_return) {
+            return $output;
+        } else {
+            return ($live_edit ? '<div id="'.$hook_name.'" class="dndHook" style="min-height:50px">' : '').$output.($live_edit ? '</div>' : '');
+        }// Return html string
+    }
+
+    /**
+     * Regenerate content of hook by adding html identifier
+     *
+     * @param string $display Get content for this hook
+     * @param string $moduleInstance Get modules content
+     * @param int $id_hook Get hook id
+     * @return string
+     */
+    public static function wrapLiveEdit($display, $moduleInstance, $id_hook)
+    {
+
+        $context = Context::getContext();
+
+        $link_module = $context->link->getBaseLink() . Tools::getValue('ad') . '/index.php?controller=AdminModules&configure=' . Tools::safeOutput($moduleInstance->name) . '&token=' . Tools::getAdminToken('AdminModules'.(int)Tab::getIdFromClassName('AdminModules').(int)Tools::getValue('id_employee'));
```

Comment:
Please use `sprintf()` to improve readability and reduce lines length (even better if you name the placeholders)